### ISSUE TITLE
API should not specifically handle Secure cookies

### ISSUE THREAD

**Issue by metromoxie** [WICG/cookie-store#20]
In Opinions, it mentions that `Secure` cookies would be treated specially by the API by not letting insecure origins touch them. I think this should be omitted. Chrome is currently releasing Strict Secure Cookies (https://www.chromestatus.com/feature/4506322921848832) which applies all of these properties to cookies across the board, including from `document.cookie`. Firefox has already started implementing this as well, and we're in touch with other browser vendors to help make it happen. Similarly, I don't think cookie prefixes should be explicitly handled. IMO, these should continue to be handled at the cookie layer in networking, and not at the Web API layer

That having been said, I like that `Secure` is the default from secure origins.

**Comment by bsittler** [comment]
Similarly to https://github.com/bsittler/async-cookies-api/issues/21#issuecomment-237400915 I think it may make sense for a browser to prevent async cookie API access or log warning messages when it is used in these situations early, since this API makes the failures much easier to detect than the pre-existing APIs do. An explicit promise rejection is much easier to debug than a silent failure, and making all the HTTP-level cookie failures from pre-existing APIs non-silent has the potential to introduce a lot more noise across the web than merely doing so for access via the new async API would.

I also think that since this is a new API, building in sane defaults and enforcing them even when the rest of the browser has not yet caught up is still sensible so that development done purely in terms of the new API starts off on the right track and is subject to fewer breaking changes in the near future.

**Comment by metromoxie** [comment]
As I mentioned offline, I'm all for promise rejection of these types of issues, rather than silent failure. However, what I'd prefer to see is somehow tying this API to use the cookie spec to say (in spec language) "for all cookie setting failures, reject the promise," and then I'd like to see non-normative language explaining what that means and why (namely, what you just wrote).

I like this approach more, because it would mean that the API spec won't need to be changed for every restriction we add to the cookie spec, which is quite likely to happen in the immediate future.

So, in short, my original request is too strongly worded. I like the promise rejection approach, I just want us to be clever in how we write the spec, and make sure we're not inadvertently changing cookie behavior, and, instead, are simply adding a sane API on top of already spec'd (or soon-to-be-spec'd) cookie behavior.

**Comment by bsittler** [comment]
Added "any cookie write operation that would result in a cookie setting failure should be rejected" and changed other rejection wording somewhat in https://github.com/bsittler/async-cookies-api/pull/23 - what do you think?

**Comment by bsittler** [comment]
I'll go ahead and merge the new language in its current form given the absence of further comments, but I'm happy to modify the explainer further if you have changes you would like to see.

**Comment by pwnall** [comment]
Closing this because it seems like it has been addressed.

@metromoxie Your feedback / help on the spec is welcome! Please reach out to me if you'd like to be involved, to make sure we ping you at the right times.
